extensions/GlobalMessages/classes/GlobalMessagesService.class.php

```diff
@@ -7,7 +7,7 @@
 class GlobalMessagesService {
 	const I18N_FILE_EXTENSION = ".i18n.php";
 	const PHP_FILE_EXTENSION = ".php";
-	const CORE_LOCALISATION_FILE_REGEX = "/^.+GlobalMessages[A-Z]+\.i18n\.php$/i";
+	const CORE_LOCALISATION_FILE_REGEX = "/^.+GlobalMessages[0A-Z]+\.i18n\.php$/i";
```

Comment:
The order in which `RebuildLocalisationCache` script gets the list of i18n files makes me confused here:

```php
$wgExtensionMessagesFiles = array_unique(
  			array_merge(
    				GlobalMessagesService::getInstance()->getCoreMessageFiles(),
  			  	GlobalMessagesService::getInstance()->getExtensionMessageFiles(),
  				  $wgExtensionMessagesFiles
  			)
		);
```

Doesn't that mean that GlobalMessages i18n files (which are meant to override the extensions i18n files) will actually be overridden by extensions?

Example from @wladekb 

```
wladek@cron-s1:~/dev/app/maintenance/wikia$ git grep wikiasearch2-search-all-wikia ../..
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Search all of Fandom',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Missing documentation',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Alle Fandom-Communitys durchsuchen',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Buscar en todo Fandom',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => "Rechercher sur l'ensemble de Fandom",
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Cerca in tutto Fandom',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'すべてのFandomから検索',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Przeszukaj cały Fandom',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Pesquisar todo o Fandom',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => 'Поиск в Фэндоме',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => '搜索所有Fandom社区',
../../extensions/GlobalMessages/GlobalMessages0.i18n.php:   'wikiasearch2-search-all-wikia' => '搜索所有Fandom社區',
../../extensions/GlobalMessages/tests/GlobalMessages0.data.php:'wikiasearch2-search-all-wikia' => 'Search all of Fandom',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Search all of Wikia',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Durchsuche alle Wikia-Wikis',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Buscar en todo Wikia',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Rechercher sur tout Wikia',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Procurar en toda Wikia',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Cerca in tutta Wikia',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Пребарај ја целата Викија',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Cari di seluruh Wikia',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Søk i hele Wikia',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Heel Wikia dooezoeken',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Szukaj na wszystkich wiki',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Поиск по всей Викия',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Säik in aal Wikia-Wikis',
../../extensions/wikia/Search/WikiaSearch.i18n.php: 'wikiasearch2-search-all-wikia' => 'Sök på hela Wikia',
../../extensions/wikia/Search/templates/WikiaSearch_CrossWiki_index.php:                            href="<?= preg_replace('/&hub=[^&]+/', '', $_SERVER['REQUEST_URI']) ?>"><?= wfMsg('wikiasearch2-search-all-wikia') ?></a>
../../extensions/wikia/Search/templates/WikiaSearch_index.php:                          <a href="<?=preg_replace( '/&hub=[^&]+/', '', $_SERVER['REQUEST_URI'] )?>"><?= wfMsg( 'wikiasearch2-search-all-wikia' ) ?></a>
```